src/pages/home/home.ts

```diff
@@ -286,86 +291,95 @@ export class HomePage {
             .navigate(
               [
                 params.selectedPosition.latitude,
-                params.selectedPosition.longitude,
+                params.selectedPosition.longitude
               ],
               {
                 app: app,
                 start: [
                   params.currentPosition.latitude,
-                  params.currentPosition.longitude,
-                ],
+                  params.currentPosition.longitude
+                ]
               }
             )
             .then(
-              success => alert('Map launching...'),
-              error => alert('Maps application failed to open!')
+              success => alert("Map launching..."),
+              error => alert("Maps application failed to open!")
             );
         });
     }
   }
 
   showCreateSpaceModal() {
-    //On open space modal, subscribe to group uid from server.
-    this.groupSocketService.uid.subscribe(
-      //User's info object that will be sent to server.
-      group_uid => {
-        this.groupSocketService.userInfo = {
-          socketID: '',
-          groupUID: group_uid,
-          username: this.username,
-          latitude: this.latitude,
-          longitude: this.longitude,
-        };
-
-        console.log(group_uid);
-
-        //Join the room specified by the group uid.
-        this.groupSocketService.joinGroup();
-
-        //Create modal.
-        let spaceModal = this.modalCtrl.create(SpacePage, {
-          uid: group_uid,
-        });
-
-        spaceModal.present();
-
-        spaceModal.onDidDismiss(data => {
-          this.groupSocketService.userInfos = [];
-        });
-      },
-      error => {
-        //Create modal.
-        let spaceModal = this.modalCtrl.create(SpacePage, {
-          uid: '',
-        });
-
-        spaceModal.present();
-
-        spaceModal.onDidDismiss(data => {
-          this.groupSocketService.userInfos = [];
-        });
+    let preferencesModal = this.modalCtrl.create(PreferencesPage);
+    preferencesModal.present();
+    preferencesModal.onDidDismiss(preferences => {
+      //If the next button was clicked, preferences were passed.
+      if (preferences != undefined) {
```

Comment:
`!preferences` should work the same way. also you should always use `===` or `!==`